src/store/ChartAdapterStore.ts

```diff
@@ -216,15 +248,87 @@ export default class ChartAdapterStore {
         }
     }
 
-    onWheel = (e: WheelEvent) => {
-        e.preventDefault();
+    onTouchStart(e: TouchEvent) {
+        const chartNode = this.mainStore.chart.chartNode;
+        if (chartNode && !this.mainStore.state.isVerticalScrollEnabled && e.touches.length === 1) {
+            const { screenX, screenY } = e.touches[0];
+            this.touchValues = { x: screenX, y: screenY };
+        }
+    }
+    onTouchEnd(e: TouchEvent) {
+        const chartNode = this.mainStore.chart.chartNode;
+        if (chartNode && !this.mainStore.state.isVerticalScrollEnabled && e.touches.length === 1) {
+            this.touchValues = { yOnTouchEnd: e.touches[0].screenY };
+        }
+    }
+
+    get scrollableChartParent() {
+        const chartNode = this.mainStore.chart.chartNode;
+        if (!chartNode) return undefined;
+        let parent = chartNode.parentElement;
+        while (parent) {
+            const { overflow } = window.getComputedStyle(parent);
+            if (overflow.split(' ').every(o => o === 'auto' || o === 'scroll')) {
+                return parent;
+            }
+            parent = parent.parentElement;
+        }
+        return document.documentElement;
+    }
+
+    onWheel = (e: WheelEvent & TouchEvent) => {
+        if (e.type === 'wheel') e.preventDefault();
+
+        // Prevent vertical scroll on the chart if isVerticalScrollEnabled is false:
+        const chartNode = this.mainStore.chart.chartNode;
+        if (chartNode && !this.mainStore.state.isVerticalScrollEnabled) {
+            const nonScrollableAreaWidth = chartNode.offsetWidth - this.mainStore.chart.yAxisWidth;
+            const { left } = chartNode.getBoundingClientRect();
+
+            if (e.type === 'touchmove') {
+                // Touch devices approach - forcing scroll on a scrollable parent of the chart:
```

Comment:
i can move this logic for touch devices to its own callback (onTouchMove). what do you think?